raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -260,6 +281,8 @@ private void maybeBecomeLeader(CandidateState state) throws IOException {
     }
 
     private void becomeCandidate() throws IOException {
+        // after we become candidate, we may not fetch from leader any more
+        fetchTimer.reset(fetchTimeoutMs);
```

Comment:
If we are a candidate, I thought we would be governed by the electionTimer. This is sort of making me wonder if we should just use a single timer and let the state dictate whether it is set to the fetch or election timeouts. Not sure if that would make the code easier to understand or not.